src/sensai/featuregen.py

```diff
@@ -104,7 +104,7 @@ def getGeneratedColumnNames(self) -> Optional[List[str]]:
         return self._generatedColumnNames
 
     @abstractmethod
-    def fit(self, X: pd.DataFrame, Y: pd.DataFrame, ctx=None):
+    def fit(self, X: pd.DataFrame, Y: pd.DataFrame = None, ctx=None):
```

Comment:
I was thinking that the next step would be to allow for the abstract subclasses of VectorModel to also be unsupervised. The actual implementations of supervised models can forbid passing None in _fit. As for feature generators, I do think it is very natural to allow passing None for Y. Even when they are fit on input data it does not necessarily need to involve the target.

I could make an UnsupervisedModel, of course, but it would involve duplicating some code or overwriting the fit (or some more serious refactoring). Also, if we were to follow my plan above, it would make the hierarchy more complicated, as we would need to duplicate all child classes. We then have an UnsupervisedClassificationVectorModel, UnsupervisedRegressionVectorModel, and so on differing only by whether Y is passed or not. 

Maybe we can have a short sync on the phone about the best way to include unsupervised models and generators?

Reply:
Yeah, I haven't completely thought this through. I reverted the relaxation of the interface in the models. As discussed, I left the relaxed interface for the feature generators, where it does make sense